server/src/main/kotlin/fi/oph/ludos/RoleUtils.kt

```diff
@@ -0,0 +1,53 @@
+package fi.oph.ludos
+
+import fi.oph.ludos.cas.Kayttajatiedot
+import org.springframework.core.env.Environment
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.security.core.context.SecurityContextHolder
+
+enum class Role {
+    YLLAPITAJA, OPETTAJA, LAATIJA, UNAUTHORIZED
+}
+
+object RoleChecker {
+    fun hasRole(permission: String, environment: Environment) = permission == getRole(environment).toString()
+
+    fun getRole(environment: Environment): Role {
+        if ("local" in environment.activeProfiles) {
+            return Role.YLLAPITAJA
+        }
+
+        val userDetails = getUserDetails()
+        val userRights = userDetails.organisaatiot.flatMap { it.kayttooikeudet }.filter { it.palvelu == "LUDOS" }
+
+        val roleMapping = mapOf(
+            "LUKU_MUOKKAUS_POISTO" to Role.YLLAPITAJA, "LUKU" to Role.OPETTAJA
+        )
+
+        return userRights.firstOrNull()?.oikeus?.let { roleMapping[it] } ?: Role.UNAUTHORIZED
```

Comment:
Tää näyttäis nyt olettavan, että käyttäjällä ei oo yhtään LUDOS-roolia, tai sit on tasan yksi.

Mitä kannattais tehdä, jos käyttäjälle on vahingossa annettu useampi rooli? Pitäiskö valita käyttäjän rooleista väkevin, ja loggaa warning?